tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc

```diff
@@ -110,6 +112,33 @@ class TrtShapeOptimizationProfileTest : public ::testing::Test {
     network->markOutput(*output);
   }
 
+  void CheckProfile(const std::vector<nvinfer1::Dims3>& dimvec,
+                    TrtShapeOptimizationProfile* profile, bool has_prof,
+                    bool test_optimality) {
+    std::vector<TensorShape> shape_vec = DimVecToShapeVec(dimvec);
+    int idx = profile->GetProfileNumber(shape_vec);
+    ASSERT_EQ(idx >= 0, has_prof);
+    if (idx < 0) return;
+    int prof_idx = exec_contexts_[idx]->getOptimizationProfile();
```

Comment:
This API is not available in TRT 5.
There are other similar errors when running the test with TRT 5:

third_party/tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc:122:41: error: no member named 'getOptimizationProfile' in 'nvinfer1::IExecutionContext'
    int prof_idx = exec_contexts_[idx]->getOptimizationProfile();
                   ~~~~~~~~~~~~~~~~~~~~~^
third_party/tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc:126:34: error: no member named 'OptProfileSelector' in namespace 'nvinfer1'
          j, prof_idx, nvinfer1::OptProfileSelector::kMIN);
                       ~~~~~~~~~~^
third_party/tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc:125:36: error: no member named 'getProfileDimensions' in 'nvinfer1::ICudaEngine'; did you mean 'getBindingDimensions'?
      nvinfer1::Dims min = engine->getProfileDimensions(
                                   ^~~~~~~~~~~~~~~~~~~~
                                   getBindingDimensions
blaze-out/k8-cuda101-opt/bin/third_party/tensorrt/_virtual_includes/nv_infer_static_cuda_10_1/third_party/tensorrt/NvInfer.h:4985:18: note: 'getBindingDimensions' declared here
    virtual Dims getBindingDimensions(int bindingIndex) const = 0;
                 ^
third_party/tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc:128:34: error: no member named 'OptProfileSelector' in namespace 'nvinfer1'
          j, prof_idx, nvinfer1::OptProfileSelector::kMAX);
                       ~~~~~~~~~~^
third_party/tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc:127:36: error: no member named 'getProfileDimensions' in 'nvinfer1::ICudaEngine'; did you mean 'getBindingDimensions'?
      nvinfer1::Dims max = engine->getProfileDimensions(
                                   ^~~~~~~~~~~~~~~~~~~~
                                   getBindingDimensions
blaze-out/k8-cuda101-opt/bin/third_party/tensorrt/_virtual_includes/nv_infer_static_cuda_10_1/third_party/tensorrt/NvInfer.h:4985:18: note: 'getBindingDimensions' declared here
    virtual Dims getBindingDimensions(int bindingIndex) const = 0;
                 ^
third_party/tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc:130:34: error: no member named 'OptProfileSelector' in namespace 'nvinfer1'
          j, prof_idx, nvinfer1::OptProfileSelector::kOPT);
                       ~~~~~~~~~~^
third_party/tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles_test.cc:129:36: error: no member named 'getProfileDimensions' in 'nvinfer1::ICudaEngine'; did you mean 'getBindingDimensions'?
      nvinfer1::Dims opt = engine->getProfileDimensions(
                                   ^~~~~~~~~~~~~~~~~~~~
                                   getBindingDimensions
blaze-out/k8-cuda101-opt/bin/third_party/tensorrt/_virtual_includes/nv_infer_static_cuda_10_1/third_party/tensorrt/NvInfer.h:4985:18: note: 'getBindingDimensions' declared here
    virtual Dims getBindingDimensions(int bindingIndex) const = 0;

Reply:
I have added an ifdef to guard the TRT6+ API calls.